### PR TITLE
Localization and String term correction

### PR DESCRIPTION
**Pull request by zora-wang**
##### 1st Commit 
The first commit is to Localize the node "Convert Between Unit ". The result is that all the ConversionUnit and ConversionMetricUnit can show in locale language.
capture

##### 2nd Commit 
* The second commit is to localize the string term " Unit ".
capture2

##### 3rd Commit
* The third commit is to correct the wrong string description for node " ByColorsAndParameters" .

##### 4th Commit
* The forth commit is to correct the description string of node" Rectangle.ByWidthLength" .
The related PR in LibG project also need to be merged : https://git.autodesk.com/Dynamo/LibG/pull/117

##### 7th Commit
* The 7th Commit is to correct wrong description of node "Math.Floor " &"Math.Ceil".

##### 8th Commit
* The 8th Commit is to correct wrong description of node "SortIndexByValue".

## Declerations
Localization issue.

## Reviewer
@ke-yu  PTAL

### CODE REVIEW

**Comment by ke-yu** [comment]
@zora-wang , please update PR title.

**Comment by zora-wang** [comment]
@ke-yu ,PTAL ,THX

File: src/Libraries/CoreNodeModelsWpf/Converters/UnitToTextConverter.cs [hunk 1]
```diff
@@ -11,11 +11,17 @@ public class UnitToTextConverter : IValueConverter
         // parameter is the data context
         // value is the selection
         public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
-        {
+        {
```
Comment (ke-yu) [inline src/Libraries/CoreNodeModelsWpf/Converters/UnitToTextConverter.cs]:
```
string unitString = string.empty;
if (value is ConversionUnit)
    unitString = (value as ConversionUnit).ToString();
else if (value is ConversionMetricUnit)
    unitString = (value as ConversionMetricUnit).ToString();

string localizedUnit = Resources.ResourceManager.GetString("Unit" + unitString, CultureInfo.CurrentUICulture);
return localizedUnit ?? string.Empty;
```

**Comment by ke-yu** [comment]
@zora-wang , if you only update protogeometry.xml, then no need to include protogeometry.dll.

File: src/Libraries/CoreNodeModels/Selection.cs [hunk 2]
```diff
@@ -102,8 +102,12 @@ public virtual string SelectionSuggestion
...
             this.selectionType = selectionType;
             this.selectionObjectType = selectionObjectType;
-
-            OutPortData.Add(new PortData("Elements", Resources.SelectionPortDataResultToolTip));
+            if (selectionMessage.Equals("Select Model Element"))
+            {
+                OutPortData.Add(new PortData("Element", Resources.SelectionPortDataResultToolTip));
+            }
+            else 
+                OutPortData.Add(new PortData("Elements", Resources.SelectionPortDataResultToolTip));
```
Comment (ke-yu) [inline src/Libraries/CoreNodeModels/Selection.cs]: 1) I didn't see any difference between if and else statement
2) what does `selectionMessage` come from? Compare with hard coded string is not reliable.

Comment (zora-wang) [inline src/Libraries/CoreNodeModels/Selection.cs]: 1. in if else statement :one is "Element",the other  is "Elements". It is to create different outports.

2.selectionMessage is from the SelectionBase, which is a parameter in SelectionBase method.

Comment (ke-yu) [inline src/Libraries/CoreNodeModels/Selection.cs]: Why this string isn't localized? 

I'd suggest to rewrite it in the other way:
```
string portName = condition ? "Element" : "Elements";
OutPortData.Add(new PortData(portName), Resources.SelectionPortDataResultToolTip));
```

Comment (zora-wang) [inline src/Libraries/CoreNodeModels/Selection.cs]: You want this ? 
string portName = selectionMessage.Equals("Select Model Element")? "Element" : "Elements";
OutPortData.Add(new PortData(portName), Resources.SelectionPortDataResultToolTip));

Comment (ke-yu) [inline src/Libraries/CoreNodeModels/Selection.cs]: Don't check the message of "Select Model Element", but check the type of node.

**Pull request merged**